File: src/Repository/RepositoryFramework.Abstractions/Consumers/Interfaces/IQuery.cs
namespace RepositoryFramework
{
    /// <summary>
    /// Interface for your CQRS pattern, with Get, Query, Operation (like Count, Sum, Max) and Exist methods.
    /// This is the interface that you need to extend if you want to create your query pattern.
    /// </summary>
    /// <typeparam name="T">Model used for your repository.</typeparam>
    /// <typeparam name="TKey">Key to retrieve your data from repository.</typeparam>
    public interface IQuery<T, TKey> : IQueryPattern<T, TKey>
        where TKey : notnull
    {

    }
}

File: src/Repository/repositoryframework.api.client.typescript/src/rystem/src/interfaces/IQuery.ts
import { QueryBuilder } from "../engine/query/QueryBuilder";
import { State } from "../models/State";

export interface IQuery<T, TKey> {
    get(key: TKey): Promise<T>;
    exist(key: TKey): Promise<State<T, TKey>>;
    query(): QueryBuilder<T, TKey>;
}

export interface IQueryPattern {
    get(key: any): Promise<any>;
    exist(key: any): Promise<State<any, any>>;
    query(): QueryBuilder<any, any>;
}
